Tests/Pcap++Test/Tests/TcpReassemblyTests.cpp

```diff
@@ -120,6 +121,40 @@ static void tcpReassemblyMsgReadyCallback(int8_t sideIndex, const pcpp::TcpStrea
 }
 
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// tcpReassemblyManuallyCloseConnMsgReadyCallback()
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+static void tcpReassemblyManuallyCloseConnMsgReadyCallback(int8_t sideIndex, const pcpp::TcpStreamData& tcpData, void* userCookie)
+{
+	TcpReassemblyMultipleConnStats::Stats &stats = ((TcpReassemblyMultipleConnStats*)userCookie)->stats;
+
+	auto iter = stats.find(tcpData.getConnectionData().flowKey);
+	if (iter == stats.end())
+	{
+		stats.insert(std::make_pair(tcpData.getConnectionData().flowKey, TcpReassemblyStats()));
+		iter = stats.find(tcpData.getConnectionData().flowKey);
+	}
+
+	iter->second.totalMissingBytes += tcpData.getMissingByteCount();
+
+	if (sideIndex != iter->second.curSide)
+	{
+		iter->second.numOfMessagesFromSide[sideIndex]++;
+		iter->second.curSide = sideIndex;
+	}
+
+	((TcpReassemblyMultipleConnStats *)userCookie)->timestamps.push_back(tcpData.getTimeStamp());
+	iter->second.numOfDataPackets++;
+	iter->second.reassembledData += std::string((char*)tcpData.getData(), tcpData.getDataLength());
+
+	// if numOfDataPackets hits 10, close the connection manually
+	if (iter->second.numOfDataPackets >= 10)
+	{
+		((TcpReassemblyMultipleConnStats*)userCookie)->tcpReassmbly->closeConnection(tcpData.getConnectionData().flowKey);
```

Comment:
Nit, ditto: C style cast

Reply:
Updated in this commit: https://github.com/seladb/PcapPlusPlus/pull/1346/commits/1aa0df801960052b3d2acc9b6e4450d16fddb728